crates/polars-core/src/chunked_array/list/mod.rs

```diff
@@ -93,3 +112,32 @@ impl ListChunked {
         })
     }
 }
+
+pub struct OffsetsIterator<'a, N>
+where
+    N: Iterator<Item = std::slice::Iter<'a, i64>>,
```

Comment:
The `slice::Iter` is a known/ easy type, so we define that one. Now we are only generic over `N`.

Reply:
Thanks for helping out on this one!